src/actions/quotes-actions.js

```diff
@@ -0,0 +1,21 @@
+import {
+  fetchQuotesBegin,
+  fetchQuotesSuccess,
+  fetchQuotesFailure,
+  getRandomQuote,
+} from './action-creators';
+import { fetchQuotes } from '../api';
+
+export const fetchQuotesAction = () => {
+  return async dispatch => {
```

Comment:
Use async/await in place of promises, Do computation in actions instead of reducer.
Reducer will only map your state (in a one-to-one fashion)